sql/core/src/test/scala/org/apache/spark/sql/ApproximatePercentileQuerySuite.scala

```diff
@@ -279,4 +282,11 @@ class ApproximatePercentileQuerySuite extends QueryTest with SharedSQLContext {
       checkAnswer(query, expected)
     }
   }
+
+  test("SPARK-24013: unneeded compress can cause performance issues with sorted input") {
+    failAfter(30 seconds) {
```

Comment:
this test looks pretty weird. Can we add some kind of unit test and move this test to PR description and say the perf has improved a lot after this patch?

Reply:
I agree that this is not the best UT, but I couldn't find any better way to test this. If anybody has any idea of a better test, I am happy to follow your right suggestion...

Reply:
We can add a UT for `ApproximatePercentile`, and check that after calling `add`, `isCompressed` is still false.